### PR TITLE
Clear cache of listing when plan is updated

### PR DESCRIPTION
**Pull request by theRealRizeo**
Fixes https://github.com/Strategy11/BusinessDirectoryPlugin/issues/4981

### CODE REVIEW

File: includes/class-listing.php [hunk 1]
```diff
@@ -636,7 +636,7 @@ public function get_fee_plan() {
 		$res = WPBDP_Utils::check_cache(
 			array(
 				'cache_key' => 'listing_fee_plan' . $this->id,
-				'group'     => 'wpbdp_listings',
+				'group'     => 'wpbdp_listing',
```
[the remark this answers is not in the document]
Comment (stephywells) [reply includes/class-listing.php]: Thanks!

**Comment by garretlaxton** [comment]
I am still seeing this message when I edit the listing and try to add an image on the front-end.
Screenshot 2021-10-20 at 16-47-03 Add Listing – Just another WordPress site

**Comment by theRealRizeo** [comment]
@garretlaxton please confirm :)

**Comment by garretlaxton** [comment]
@theRealRizeo Here is a quick video of what happens from my end: https://www.loom.com/share/30841aaa9fe04b5ab4d2fed7114fda57

**Comment by theRealRizeo** [comment]
@garretlaxton thanks, please test with the latest changes. Both for the current issue and https://github.com/Strategy11/BusinessDirectoryPlugin/issues/4986

**Comment by garretlaxton** [comment]
@theRealRizeo Unfortunately, this didn't fix either issue.

**Comment by stephywells** [comment]
@theRealRizeo Looks like this PR is still in progress.

**Comment by theRealRizeo** [comment]
Fix works on automated tests and on local tests. @garretlaxton could you please try the same on the public QA site?

**Comment by garretlaxton** [comment]
@theRealRizeo I pushed this to the QA site & still have the issue. You can see the listing here: https://qa.businessdirectoryplugin.com/business-directory/business-directory/

Edit the listing from the frontend and try to add another image, the slots are full warning shows.

**Comment by theRealRizeo** [comment]
Thanks, seen the issue

**Comment by theRealRizeo** [comment]
Okay, I have seen the issue and might need some advise before changing it. 
@stephywells the way it currently works is a listing that is created with a particular plan, will keep that plan settings at that current time. If the admin of the site decides to change the plan to more images, users who were using the previous version of the plan will not get these new changes.
In this case if a listing is created with a plan for 2 images, then later the admin upgrades the plan to 10 images, the old listing will not get to use the 10 images unless manually updated by the admin in the backend

**Comment by stephywells** [comment]
@theRealRizeo I could see this going both ways. So how about this?

After a plan is updated, we can check if the image limit changed, and if there are any listings for that plan. If there are, let's show a link in the success message the update limits on existing listings.

But I would say this is an enhancement rather than a bug since expectations could be different in different situations.

**Comment by theRealRizeo** [comment]
@stephywells that makes sense. Will close this and update the Bug report with your suggestion :)

**Pull request closed without merging**